src/snowflake/snowpark/files.py

```diff
@@ -122,6 +125,20 @@ def isatty(self) -> None:
         """
         raise NotImplementedError(_DEFER_IMPLEMENTATION_ERR_MSG)
 
+    @classmethod
+    @private_preview(version="1.22.1")
+    def open_new_result(cls, mode: str = "w") -> SnowflakeFile:
+        """
+        Returns a :class:`~snowflake.snowpark.file.SnowflakeFile`.
+        In UDF and Stored Procedures, the object works like a Python IOBase object and as a wrapper for an IO stream of remote files. The IO Stream is to support the file operations defined in this class.
+
+        You can use this particular scoped URL to access files from within Snowflake (through another UDF or the COPY FILES command), but not from outside of Snowflake as a pre-signed URL. The scoped URL is valid for 24 hours.
+
+        Args:
+            mode: A string used to mark the type of an IO stream.
+        """
+        return cls("new results file", mode, require_scoped_url=0, from_result_api=True)
```

Comment:
if we always write the result in one location - `"new results file"`, we should probably mention that in the documentation.

Reply:
yes, but it's more like a placeholder for the file_uri. also user should read/write file through scoped url, maybe it's better to hide this info to user?

Reply:
Hrm, reading through the comments. The user of this class doesn't need to do anything with scoped URLs. Probably would be best not to mention that and instead do something like:

"This stream will open a writable result file that will be materialized when it's returned by a UDF or Stored Procedure. When the file is materialized then file_uri will be set to a scoped URL that temporarily references this file."